### PR TITLE
Feature: pre-commit check hex case

### PR DESCRIPTION
**Pull request by perigoso**
This PR adds a hook to the pre-commit file, said hooks checks for hex number declarations (i.e. `0xffff`) case (as in upper or lower case) across the committed files

~edit: an additional hook to check for accidental assignment has been added~

## Detailed description

It is set to check and fix declarations to the format `0x55aa`, i.e. all lower case

note that this depends on the merge of #1282 as it includes the addition of the pre-commit file, it also should only be commited after the code has been cleaned up (as in the hex number declarations all fixed to consistent format)

@dragonmux or anyone that wants to make use of the hook before it is configured (i.e. this has been merged), you can clone the hook repository and use it as a python script:

`git clone https://github.com/perigoso/pre-commit-hooks.git`

`${path_to_folder}/pre-commit-hooks/hooks/check_hex_case.py -i --lower-digits ${file_to_fixup}`

this will fix all hex numbers in the passed file, you can pass multiple files, that should prove useful for the cleanup

## Your checklist for this pull request

* [x] I've read the [Code of Conduct](https://github.com/blackmagic-debug/blackmagic/blob/main/CODE_OF_CONDUCT.md)
* [x] I've read the [guidelines for contributing](https://github.com/blackmagic-debug/blackmagic/blob/main/CONTRIBUTING.md) to this repository
* [x] It builds for hardware native (`make PROBE_HOST=native`)
* [x] It builds as BMDA (`make PROBE_HOST=hosted`)
* [x] I've tested it to the best of my ability
* [x] My commit messages provide a useful short description of what the commits do

### CODE REVIEW

**Review by dragonmux — DISMISSED** [review]
This LGTM, we'll merge it once the various pending cleanup branches have been merged as this currently makes the linting step very shouty, which should reduce considerably as those get merged.

**Comment by dragonmux** [comment]
The cleanup branches have now all been merged, please rebase this against main.

**Comment by perigoso** [comment]
There were a lot of touchups needed to get a pass on these hooks, and some of the accidental attribution changes were harder to make so I haven't gone through all of them yet, I will open a separate PR for that

File: src/target/ch32f1.c [hunk 1]
```diff
@@ -41,18 +41,18 @@ static bool ch32f1_flash_write(target_flash_s *f, target_addr_t dest, const void
 #define FPEC_BASE     0x40022000
```
Comment (dragonmux) [inline src/target/ch32f1.c:41]: All these want `U` on the end

File: src/target/lpc11xx.c [hunk 2]
```diff
@@ -79,61 +79,61 @@ static bool lpc11xx_detect(target_s *const t)
...
-	case 0x2532902B: /* LPC1113/201/202 - 24K Flash 4K SRAM */
-	case 0x0434102B: /* LPC1113/301 - 24K Flash 8K SRAM */
-	case 0x2532102B: /* LPC1113/301/302 - 24K Flash 8K SRAM */
-	case 0x0A40902B: /* LPC1114/102 - 32K Flash 4K SRAM */
-	case 0x1A40902B: /* LPC1114/102 - 32K Flash 4K SRAM */
-	case 0x0444502B: /* LPC1114/201 - 32K Flash 4K SRAM */
-	case 0x2540902B: /* LPC1114/201/202 - 32K Flash 4K SRAM */
-	case 0x0444102B: /* LPC1114/301 - 32K Flash 8K SRAM */
-	case 0x2540102B: /* LPC1114/301/302 & LPC11D14/302 - 32K Flash 8K SRAM */
+	case 0x0a07102b: /* LPC1110 - 4K Flash 1K SRAM */
```
Comment (dragonmux) [inline src/target/lpc11xx.c:90]: These really should all be suffixed with `U`

File: src/target/lpc43xx.c [hunk 3]
```diff
@@ -80,40 +80,40 @@ bool lpc43xx_probe(target_s *t)
 	chipid = target_mem_read32(t, LPC43XX_CHIPID);
 
 	switch (chipid) {
-	case 0x4906002B: /* Parts with on-chip flash */
-	case 0x7906002B: /* LM43S?? - Undocumented? */
-		switch (t->cpuid & 0xFF00FFF0) {
-		case 0x4100C240:
+	case 0x4906002b: /* Parts with on-chip flash */
```
Comment (dragonmux) [inline src/target/lpc43xx.c:83]: We would say "Add `U`" here, but that fells unnecessary when v1.10 will see this code replaced wholesale when we land feature/lpc4370-external-flash-support as we rewrote such a large part of this file.

File: src/target/lpc546xx.c [hunk 4]
```diff
@@ -34,10 +34,10 @@
 /* Only SRAM0 bank is enabled after reset */
 #define LPC546XX_ETBAHB_SRAM_SIZE (64 * 1024)
 
-#define LPC546XX_WDT_MODE       0x4000C000
-#define LPC546XX_WDT_CNT        0x4000C004
-#define LPC546XX_WDT_FEED       0x4000C008
-#define LPC546XX_WDT_PERIOD_MAX 0xFFFFFF
+#define LPC546XX_WDT_MODE       0x4000c000
```
Comment (dragonmux) [inline src/target/lpc546xx.c:37]: These want `U` on the end

File: src/target/nrf51.c [hunk 5]
```diff
@@ -59,11 +59,11 @@ const command_s nrf51_read_cmd_list[] = {
 };
 
 /* Non-Volatile Memory Controller (NVMC) Registers */
-#define NRF51_NVMC           0x4001E000
+#define NRF51_NVMC           0x4001e000
```
Comment (dragonmux) [inline src/target/nrf51.c:62]: `U` on the ends

File: src/target/nxpke04.c [hunk 6]
```diff
@@ -231,7 +231,7 @@ bool ke04_probe(target_s *t)
 	f->erase = ke04_flash_erase;
 	f->write = ke04_flash_write;
 	f->done = ke04_flash_done;
-	f->erased = 0xFFu;
+	f->erased = 0xffu;
```
Comment (dragonmux) [inline src/target/nxpke04.c:234]: `U` is the wrong case here

**Review by dragonmux — CHANGES_REQUESTED** [review]
This is looking good. There are a couple of small mistakes here an there with `U` being lower case still, and we're still dreadfully missing consistency with unsigned specification for unsigned constants, but we won't make that a blocker for merging this if you don't want to deal with all of those missing `U` right now.

We have also not been thorough in noting all the places where `U` is missing in the review as it's.. a lot..

**Comment by perigoso** [comment]
> 

I didn't try to be very thorough with the unsigned suffixes, because like you said, It's lot, I try and fixup at least the noted, and maybe got through a bit more thoroughly, at least fix the lower case as I might be able to hack the hex script for that

File: src/target/adiv5.c [hunk 7]
```diff
@@ -92,41 +92,41 @@ static const char *const cidc_debug_strings[] = {
...
+#define PIDR2_OFFSET 0xfe8 /* DBGPID2 */
+#define PIDR3_OFFSET 0xfec /* DBGPID3 */
+#define PIDR4_OFFSET 0xfd0 /* DBGPID4 */
+#define PIDR5_OFFSET 0xfd4 /* DBGPID5 (Reserved) */
+#define PIDR6_OFFSET 0xfd8 /* DBGPID6 (Reserved) */
+#define PIDR7_OFFSET 0xfdc /* DBGPID7 (Reserved) */
 
 #define PIDR_JEP106_CONT_OFFSET 32ULL                                /*JEP-106 Continuation Code offset */
-#define PIDR_JEP106_CONT_MASK   (0xFULL << PIDR_JEP106_CONT_OFFSET)  /*JEP-106 Continuation Code mask */
+#define PIDR_JEP106_CONT_MASK   (0xfULL << PIDR_JEP106_CONT_OFFSET)  /*JEP-106 Continuation Code mask */
```
[the remark this answers is not in the document]
Comment (perigoso) [reply src/target/adiv5.c:115]: `pidr` is a `uint64_t` and is what these macros are used in, hence these being `ULL`, I will change it to `UINT64_C` does that sound good?

[the remark this answers is not in the document]
Comment (perigoso) [reply src/target/adiv5.c:115]: hm, the compiler doesn't like this for some reason

File: src/target/nxpke04.c (hunk 6 above)
Comment (perigoso) [reply src/target/nxpke04.c:234]: According to my script, this was the last instance of lower case unsigned suffix :tada:

File: src/target/adiv5.c (hunk 7 above)
[the remark this answers is not in the document]
Comment (dragonmux) [reply src/target/adiv5.c:115]: Yes, ULL is not guaranteed to be correct for that purpose, UINT64_C() is 👍🏼

**Comment by perigoso** [comment]
This PR suffered from a bit of scope creep, and is now slightly massive, sorry

**Comment by perigoso** [comment]
This PR suffered from a bit of scope creep, and is now slightly massive, sorry

It now encompasses a full numeric constants cleanup across the code base

**Comment by dragonmux** [comment]
> This PR suffered from a bit of scope creep, and is now slightly massive, sorry

Not a problem

> It now encompasses a full numeric constants cleanup across the code base

Thank you, we'll do a review shortly and look at merging this ASAP. That should then leave your other feature branch and the RP2040 fix branch as the last things that need merging prior to v1.9rc1

**Review by dragonmux — APPROVED** [review]
Looks like we're still have a little ways to go till we're at complete consistency on how constants are specified and signed-unsigned conversions, but this is so much better that we'll go with it as-is anyway. Many thanks for your hard work on this!

**Pull request merged**